### ISSUE TITLE
Make it possible to upload and incorporate resourcemetadata.xml files created by others

### ISSUE THREAD

**Issue by alvacouch** [hydroshare/hydroshare#1598]
@horsburgh @dtarb 

Currently, uploading of resourcemedata.xml as a file doesn't work reliably. Part of the reason for this is that the structure of the serializer for resourcemetadata.xml is difficult to unserialize. 

Some definitions: in Django: 

* **Serialization** is the process of converting a Django object to a string that can be sent in a REST request. 
* **Unserialization** is the process of converting from a string (including XML) to a Django object. 

The problem is to be able to serialize from Django BaseResource into resourcemetadata.xml, and then to be able to deserialize any reasonable resourcemetadata.xml back into a BaseResource. This can be made to work reliably, but really getting it to work requires some changes in architecture:

* Currently, the serializer for resourcemetadata.xml is somewhat monolithic, in the sense that the serializer consists of large blocks of python code. Further, the current serializers do not utilize state of the art RDF libraries that make unserialization easier. 
* Django allows implementations of object serializers from field serializers, in smaller pieces that are easier to test for proper serialization and unserialization. 
* A serializer is **reflexive** if serialize(unserialize(X)) = X (for X a string) and unserialize(serialize(Y)) for Y an object. This means that one can upload something and then download it, or download it and then upload it with robust and expected results. 

The key to the re-architecting is that: 

* A Django serializer can be composed of smaller serializers that are easy to make work, and 
* **A composite Django serializer is reflexive if all of its component serializers (field and object) are themselves reflexive.** 

In layman's terms, we want to create a composite serializer/deserializer pair where the input to the serializer is a Django object and the output is RDF/XML that we store in a file. 

Steps for this are as follows: 

1. Create reflexive serializers for each field in BaseResource. These should write RDF/XML and utilize python-rdflib for ease of programming. 
2. Ensure that each of these is reflexive and can be read back into a Django object correctly by implementing an unserialize method for the RDF/XML, also using python-rdflib. 
3. Create a composite serializer that calls all of these to create the whole resourcemetadata file. And then the composite serializer will be reflexive and provide an unserializer method that will be able to ingest conforming resourcemetadata.xml files! 

It is a total bear to write a monolithic unserializer for our current serializer. Writing these "little" serialize/unserialize pairs and then composing them is considerably easier. 

Note that: 
1. **The format or resourcemetadata.xml will not change.** 
2. **The format of ingested resourcemetadata.xml will allow order differences that are not reflected in the standardized serialization.** So, one can be a bit sloppy and the unserializer will clean it up.

**Comment by alvacouch** [comment]
The reason that I claimed that this is related to issue #1575 is that enabling partial upload of resourcemetadata.xml requires the same kind of serialization, but in XML, rather than RDF. Fortunately, in that case, Django writes most of these for us.

**Comment by Castronova** [comment]
@sblack-usu Is this still relevant given your work on DSAW? If not, please close.

**Comment by sblack-usu** [comment]
fixed in #3928
